app/scrapers/generalized.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import print_function
+import os, json, sys
+import requests
+from bs4 import BeautifulSoup
+
+defaultHeaders = {
+    'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/27.0.1453.116 Safari/537.36'
+}
+
+class Scraper:
+    """Generalized scraper"""
+    url = ''
+    startKey = 'start'
+
+    def __init__(self):
+        pass
+
+    def get_page(self, query, startIndex = 0):
+        """ Fetch the google search results page
+        Returns : Results Page
+        """
+        payload = {'q': query, self.startKey: startIndex}
+        response = requests.get(self.url, headers=defaultHeaders, params=payload)
+        return response
+        
+    def parseResponse(self, soup):
+        raise NotImplementedError
+        
+    def nextStart(self, currentStart, prevResults):
+        return currentStart + len(prevResults)
+
+    def search(self, query, numResults = 10):
+        """
+            Search for the query and return set of urls
+            Returns: list
+        """
+        urls = []
+        currentStart = self.defaultStart
+
+        while(len(urls) < numResults):
```

Comment:
This may cause problems @AnshulMalik 
If number of results are never greater than numResults then the scrapper may get stuck in the while loop and will generate 500 error.  So this can be replaced by the previous version with a seperate if command for length check and a return statement outside search loop to return the maximum results it can generate. 
Please take a look at all places where you've done it.
Cheers.

Reply:
I'm sorry i am busy with my classes. So kindly check for it. Therefore i added all in a single comment.

Reply:
That case won't arrive here, as if number of results are less, it'll go fetch the next page

Now if we have more results, then it'll return only the required ones.

Reply:
I'm sorry Anshul but it is difficult for me to wrap my head around what you stated. If you could show me a screenshot with a query of 100 results ( because 100 results are not fetched generally, it responds with fewer queries) . Then i can approve it. Thanks.